src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -404,7 +440,7 @@ export class CommandDetectionCapability extends Disposable implements ICommandDe
 
 	handleCommandExecuted(options?: IHandleCommandOptions): void {
 		if (this._isWindowsPty) {
-			this._handleCommandExecutedWindows();
+			this._commandStartWindowsPromise?.then(() => this._handleCommandExecutedWindows());
```

Comment:
We want to "flush" this task and execute it immediately instead of wait for it as it could back up and never get canceled. You can do this in a few ways, for example using a Barrier and calling `open()` before handling execution/finished